lib/Zonemaster/CLI.pm

```diff
@@ -312,7 +315,12 @@ sub run {
     Zonemaster::Engine::Profile->effective->set( q{net.ipv6}, 0+$self->ipv6 );
 
     if ($self->sourceaddr) {
-        Zonemaster::Engine::Profile->effective->set( q{resolver.source}, $self->sourceaddr );
+        if ($self->check_sourceaddress_exists ) {
+            Zonemaster::Engine::Profile->effective->set( q{resolver.source}, $self->sourceaddr );
+        }
+	else {
+            die __( "Address '" ) . $self->sourceaddr . __("' cannot be used as source address for DNS queries.") . "\n";
```

Comment:
The gettext documentation on [preparing translatable strings](https://www.gnu.org/software/gettext/manual/html_node/Preparing-Strings.html#Preparing-Strings) states that "[T]ranslatable strings should be entire sentences", and prescribes format strings as a solution. For consistency with existing translation strings, I suggest using [perl-brace-format](https://www.gnu.org/software/gettext/manual/html_node/perl_002dformat.html#perl_002dformat) (via `__x`). 

I recommend reading the entire page on preparing translatable strings. It's not too long and it provides good understanding of the interface between programmers and translators.

Reply:
I find [preparing translatable strings] worth reading and I agree that we should follow that advice.

[preparing translatable strings]: https://www.gnu.org/software/gettext/manual/html_node/Preparing-Strings.html#Preparing-Strings

Reply:
Fixed